### DISCUSSION TITLE
Bypass VPS on selected routes

### DISCUSSION THREAD

**Discussion by brillout** [vikejs/vike#782] [Discussion] [answered]
Moved from https://github.com/brillout/vite-plugin-ssr/issues/568#issuecomment-1506504228 @TomLadek:

> > @bagr001 What's your motivation for not completely bypassing VPS and creating a normal Fastify route instead?
> 
> @brillout What would be the standard way to bypass VPS on selected routes? I have a very similar situation as bagr001 in that I want to use VPS client routing for most routes as well as some API GET routes handled (exclusively) by the server.
> 
> My current solution is to create a dummy route in `pages/` but I feel there has to be a nicer way:
> ```JavaScript
> // pages/rest/rest.page.route.js
> export default "/rest/*"
> ```
> 
> ```JavaScript
> // pages/rest/_default.page.server.js
> export async function render() {
>   return {}
> }
> ```
> 
> ```JavaScript
> // server/index.js
> app.get('/rest/*', (req) => {
>   ...
> })
> ```

**Comment by brillout** [comment]
@TomLadek Have you seen https://vite-plugin-ssr.com/api-routes?

**Reply by TomLadek** [reply]
@brillout Thanks! But adding a new dependency like Telefunc or tRPC seems like an overkill for me right now. I'll stick with the solution with the dummy route for the moment.

**Comment by samuelstroschein** [comment, marked as answer]
If you use VPS with a middleware, place the VPS middleware last to bypass the catch-all routes. 

See https://github.com/inlang/inlang/blob/f491813bdfea8856d261b6bcd8525506f153f68c/source-code/server/src/main.ts#L53. We have one server that deploys RPCs, APIs, reverse proxies and vite plugin ssr. Works fabulous!

**Reply by TomLadek** [reply]
You're right, it's just the ordering of the routes 🤦🏻I've moved the VPS catch-all-route after the `/rest` route declarations and now I don't need the dummy files in `pages/rest/`. Thanks!

**Reply by brillout** [reply]
I updated the docs https://github.com/brillout/vite-plugin-ssr/commit/4bef98318fcee49cae064b10df0fa8bff5b10a38 https://vite-plugin-ssr.com/api-routes

(@TomLadek In case you think you can convince your employer: https://github.com/sponsors/brillout. Sponsors are starting to have a tremendous impact.)
